hipamd/include/hip/amd_detail/amd_warp_functions.h

```diff
@@ -123,8 +123,8 @@ unsigned long long __activemask() {
 #endif // HIP_ENABLE_WARP_SYNC_BUILTINS
 
 __device__ static inline unsigned int __lane_id() {
-    return  __builtin_amdgcn_mbcnt_hi(
-        -1, __builtin_amdgcn_mbcnt_lo(-1, 0));
+    if (warpSize == 32) return __builtin_amdgcn_mbcnt_lo(-1, 0);
+    return __builtin_amdgcn_mbcnt_hi(-1, __builtin_amdgcn_mbcnt_lo(-1, 0));
```

Comment:
My understanding is that this optimization is already performed by the compiler so we need not do it here.  If I am mistaken, then I think the compiler should be enhanced rather than expanding this function.

Reply:
Whilst your valuable insight is appreciated, your understanding is erroneous in this case (it is true for concrete cases, not for SPIR-V). There's no enhancement to do in the compiler, and definitely not one that can be implemented rapidly enough to meet schedule.

Reply:
Thanks.  We spoke offline and my understanding has improved.  I am OK with this for now but would like to see this reverted as some point in the future.

Reply:
I still don't get it. Spirv just needs to pass through the intrinsics? Whenever it reaches final codegen the dead half will fold out in wave32